hack/deploy-karmada-by-operator.sh

```diff
@@ -0,0 +1,116 @@
+
+function usage() {
+  echo "This script deploys karmada control plane components to a given cluster via karmada-operator."
+  echo "Note: This script is an internal script and is not intended used by end-users."
+  echo "Usage: hack/deploy-karmada-by-operator.sh <KUBECONFIG> <CONTEXT_NAME> <ADDON_NEEDED> <CRD_LOCAL_PATH>"
+  echo "Example: hack/deploy-karmada.sh ~/.kube/config karmada-host true"
+  echo -e "Parameters:\n\tKUBECONFIG\t\tYour cluster's kubeconfig that you want to install to"
+  echo -e "\tCONTEXT_NAME\t\tThe name of context in 'kubeconfig'"
+  echo -e "\tADDON_NEEDED\t\tWhether you need to install addons(KarmadaSearch&karmadaDescheduler)"
+  echo -e "\tCRD_LOCAL_PATH\t\tPath to the local crds.tar.gz file, if not provided, automatically pulls up the latest crds.tar.gz file in the github repo'"
+}
+
+# recover the former value of KUBECONFIG
+function recover_kubeconfig {
+  if [ -n "${CURR_KUBECONFIG+x}" ];then
+    export KUBECONFIG="${CURR_KUBECONFIG}"
+  else
+    unset KUBECONFIG
+  fi
+}
+
+if [[ $# -lt 2 ]]; then
+  usage
+  exit 1
+fi
+
+# check config file existence
+HOST_CLUSTER_KUBECONFIG=$1
+if [[ ! -f "${HOST_CLUSTER_KUBECONFIG}" ]]; then
+  echo -e "ERROR: failed to get kubernetes config file: '${HOST_CLUSTER_KUBECONFIG}', not existed.\n"
+  usage
+  exit 1
+fi
+
+# check context existence and switch
+# backup current kubeconfig before changing KUBECONFIG
+if [ -n "${KUBECONFIG+x}" ];then
+  CURR_KUBECONFIG=$KUBECONFIG
+fi
+export KUBECONFIG="${HOST_CLUSTER_KUBECONFIG}"
+HOST_CLUSTER_NAME=$2
+if ! kubectl config get-contexts "${HOST_CLUSTER_NAME}" > /dev/null 2>&1;
+then
+  echo -e "ERROR: failed to get context: '${HOST_CLUSTER_NAME}' not in ${HOST_CLUSTER_KUBECONFIG}. \n"
+  usage
+  recover_kubeconfig
+  exit 1
+fi
+
+# deploy karmada-operator
+kind load docker-image "${REGISTRY}/karmada-operator:${VERSION}" --name="${HOST_CLUSTER_NAME}"
+helm --kubeconfig ${HOST_CLUSTER_KUBECONFIG} install karmada-operator -n karmada-system  --create-namespace --dependency-update ./charts/karmada-operator --debug
+util::wait_pod_ready "${HOST_CLUSTER_NAME}" "karmada-operator" "${KARMADA_SYSTEM_NAMESPACE}"
+# preset crds.tar.gz
+if [[ -n "$4" ]]; then
+	name=$(kubectl get pods -nkarmada-system|grep operator|awk '{print $1}')
+	kubectl exec -it $name -nkarmada-system -- mkdir -p /var/lib/karmada/"${KARMADA_VERSION}"
+	kubectl cp crds.tar.gz karmada-system/$name:/var/lib/karmada/"${KARMADA_VERSION}"
+fi
+kubectl apply -f operator/config/crds/
+kubectl create ns test
+
+# if choosing install addons, append karmadaSearch and karmadaDescheduler to 'karmada.yaml'
+if [ $3 ]; then
+    TEMP_PATH=$(mktemp -d)
+    trap '{ rm -rf ${TEMP_PATH}; }' EXIT
+    cp -rf "${REPO_ROOT}"/operator/config/samples/karmada.yaml "${TEMP_PATH}"/karmada-samples-tmp.yaml
+    echo -e '    karmadaDescheduler:\n      imageRepository: docker.io/karmada/karmada-descheduler\n      imageTag: v1.8.0\n      replicas: 1' >> "${TEMP_PATH}"/karmada-samples-tmp.yaml
```

Comment:
Can the karmada image tag change dynamically?

Reply:
Dynamic injection of image tags is a good optimization point. However, since this modification would involve some documentation changes such as：
https://github.com/karmada-io/karmada/blob/6e5a6029922091e22bdae470323c2483334273ff/operator/README.md?plain=1#L85-L89
Would it be better to submit a separate PR to solve this problem?

Reply:
ok, i'm with you